### ISSUE TITLE
Got probability map is all zero?

### ISSUE THREAD

**Issue by DISAPPEARED13** [VainF/DeepLabV3Plus-Pytorch#98]
Hi, there, I use DeeplabV3+ for a multiple class semantic segmentation. But I found that the network got all-zero probability map as a result, what's the problem it will be? After I check the middle output, I got maybe the zero-map is generated from ASPP module.
How can I edit this module to get a probable output?

And, as a result, this result not generated randomly, sometimes got normal output, but sometimes got all zero. This is very confusing...

Thanks a lot!

**Comment by DaddyWesker** [comment]
I'm getting all-zero prediction from model as well. I'm using predict.py, downloaded `best_deeplabv3plus_resnet101_voc_os16.pth` , loaded it, sending simple png image as an input and getting black output (preds are all zero). And this result is not random - i'm getting it every time.
P.S. nvm, I've tried to load another ckpt trained on cityscapes and it gaves me a result.

**Comment by DISAPPEARED13** [comment]
> I'm getting all-zero prediction from model as well. I'm using predict.py, downloaded `best_deeplabv3plus_resnet101_voc_os16.pth` , loaded it, sending simple png image as an input and getting black output (preds are all zero). And this result is not random - i'm getting it every time. P.S. nvm, I've tried to load another ckpt trained on cityscapes and it gaves me a result.

Hi, there, thanks for advice. I got that because the loss function might not suit my problem, so just check and change it will be fine.
There were two bugs in my issue, one is the loss function, the second one is the softmax-before-loss problem. After all these problems fixed my train progress went well. The loss function I use now is Dice+CrossEntropy, even only with one of them should work.
